File: core/src/main/java/net/siegerpg/siege/core/fishing/catches/loot/GoldenCarrot.java
package net.siegerpg.siege.core.fishing.catches.loot;

import net.siegerpg.siege.core.fishing.catches.Fish;

public class GoldenCarrot extends Fish {

	public GoldenCarrot () {

		super(15, 1, 28, 8,
		      new net.siegerpg.siege.core.items.implemented.misc.food.GoldenCarrot(0).getUpdatedItem(false));
	}

}

File: core/src/main/kotlin/net/siegerpg/siege/core/items/implemented/misc/food/GoldenCarrot.kt
package net.siegerpg.siege.core.items.implemented.misc.food

import net.siegerpg.siege.core.items.enums.Rarity
import net.siegerpg.siege.core.items.types.misc.CustomFood
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class GoldenCarrot() : CustomFood(
	name = "Golden Carrot",
	customModelData = 330005,
	description = listOf("Mercy on your soul"),
	levelRequirement = 0,
	material = Material.GOLDEN_CARROT, //change this to cooked chicken later
	health = 200.0
) {

	constructor(quality: Int) : this() {
		this.quality = 100
		this.rarity = Rarity.LEGENDARY
		this.serialize()
	}

	constructor(item: ItemStack) : this() {
		this.item = item
		this.deserialize()
	}

}